sympy/geometry/util.py

```diff
@@ -310,6 +312,170 @@ def _orientation(p, q, r):
     return Polygon(*convexHull)
 
 
+def closest_points(*args):
+    """The closest pair of points among a given set of points in 2D plane.
+
+    Parameters
+    ==========
+
+    args : a collection of Points on 2D plane.
+
+    Returns
+    =======
+
+    pair : A pair of closest points
+
+    Notes
+    =====
+
+    This can only be performed on a set of non-symbolic points.
+
+    References
+    ==========
+
+    [1] http://www.cs.mcgill.ca/~cs251/ClosestPair/ClosestPairPS.html
+
+    [2] Sweep line algorithm
+    https://en.wikipedia.org/wiki/Sweep_line_algorithm
+
+    Examples
+    ========
+
+    >>> from sympy.geometry import closest_points, Point2D
+    >>> points = [Point2D(5, 2), Point2D(7, 6), Point2D(3, 9), Point2D(9, 4), Point2D(8, 5)]
+    >>> closest_points(*points)
+    (Point2D(8, 5), Point2D(7, 6))
+
+    >>> from sympy.geometry import closest_points, Point2D
+    >>> points = [Point2D(80, 89), Point2D(11, 85), Point2D(32, 51), Point2D(59, 59), Point2D(39, 80), Point2D(62, 73)]
+    >>> closest_points(*points)
+    (Point2D(62, 73), Point2D(59, 59))
+
+    """
+    from .point import Point, Point2D
+    from collections import deque
+    import numbers
+    from math import sqrt, pow
+
+    points = list(args)
+    for point in points:
+        if not all([point.x.is_number, point.y.is_number]):
+            raise NotImplementedError('The input must be a set of 2D points')
+    if len(points) < 2:
+        raise NotImplementedError('At-least 2 points must be inserted')
+    p = points
+    p.sort(key=lambda x: x.args)
+
+    box = deque()
+    box.append(p[0])
+
+    best_dist = sqrt(pow(p[1].y-box[0].y, 2)+pow(p[1].x-box[0].x, 2))
```

Comment:
There is already a method for calculating distance between two points, use that.

```python
In [2]: Point2D(0, 6).distance(Point2D(8, 0))
Out[2]: 10
```